packages/contracts-periphery/contracts/L2/messaging/L2ERC721Bridge.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+/* Interface Imports */
+import { IL1ERC721Bridge } from "../../L1/messaging/IL1ERC721Bridge.sol";
+import { IL1ERC721Bridge } from "../../L1/messaging/IL1ERC721Bridge.sol";
+import { IL2ERC721Bridge } from "./IL2ERC721Bridge.sol";
+
+/* Library Imports */
+import {
+    CrossDomainEnabled
+} from "@eth-optimism/contracts/contracts/libraries/bridge/CrossDomainEnabled.sol";
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+/* Contract Imports */
+import { IL2StandardERC721 } from "../../standards/IL2StandardERC721.sol";
+
+/**
+ * @title L2ERC721Bridge
+ * @dev The L2 ERC721 bridge is a contract which works together with the L1 ERC721 bridge to
+ * enable ERC721 transitions between L1 and L2.
+ * This contract acts as a minter for new tokens when it hears about deposits into the L1 ERC721
+ * bridge.
+ * This contract also acts as a burner of the token intended for withdrawal, informing the L1
+ * bridge to release the L1 NFT.
+ */
+contract L2ERC721Bridge is IL2ERC721Bridge, CrossDomainEnabled {
+    /********************************
+     * External Contract References *
+     ********************************/
+
+    address public l1ERC721Bridge;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _l2CrossDomainMessenger Cross-domain messenger used by this contract.
+     * @param _l1ERC721Bridge Address of the L1 bridge deployed to the main chain.
+     */
+    constructor(address _l2CrossDomainMessenger, address _l1ERC721Bridge)
+        CrossDomainEnabled(_l2CrossDomainMessenger)
+    {
+        l1ERC721Bridge = _l1ERC721Bridge;
+    }
+
+    /***************
```

Comment:
Since this is going to be deployed to L2 and not as a predeploy and also behind a proxy, it will need to have an initialize function instead of having logic in the constructor

Reply:
Added the initialize function and updated tests